### DISCUSSION TITLE
start from scratch, DataWharf and PG in docker

### DISCUSSION THREAD

**Discussion by janbenda** [EricLendvai/DataWharf#31] [General]
[start_dw.sh.zip](https://github.com/EricLendvai/DataWharf/files/12259296/start_dw.sh.zip)
Hi Eric,
I am trying to start DataWharf in docker with Postgres in docker too. There is probably some issue with rights on public schema in PG15 so I am using PG14
I am using tihis script to start and error apears in browser Error: Bound error, operation: array access
and in "ORM"."SchemaAndDataErrorLog" I can see some errors like 
Auto-Casing Error: Failed To find table "FastCGIRunLog".:
SQLExec Error Code: 1 - Error description: 42P01 ERROR: relation "ORM.SchemaCacheLog" does not exist

**script for start PG and DataWharf: attached in zip format too**
POSTGRES_VERSION=14
POSTGRES_CONTAINER=test-postgres-dw
DATAWHARF_CONTAINER=test-datawharf
DATAWHARF_PG_DB=datawharfdemo
DATAWHARF_PG_USER=datawharf
DATAWHARF_PG_PASSWORD=pospas

git clone https://github.com/EricLendvai/DataWharf.git

cd DataWharf/

docker stop ${DATAWHARF_CONTAINER}; docker rm  ${DATAWHARF_CONTAINER}
docker stop ${POSTGRES_CONTAINER}; docker rm ${POSTGRES_CONTAINER}

docker run --name ${POSTGRES_CONTAINER} -p 5432 -e POSTGRES_PASSWORD=${DATAWHARF_PG_PASSWORD} -d postgres:${POSTGRES_VERSION}
sleep 10

docker exec -u postgres ${POSTGRES_CONTAINER} createuser ${DATAWHARF_PG_USER}
docker exec -u postgres ${POSTGRES_CONTAINER} createdb --owner=${DATAWHARF_PG_USER} ${DATAWHARF_PG_DB}

docker exec -u postgres ${POSTGRES_CONTAINER} psql -P pager=off -U postgres -xc "alter user ${DATAWHARF_PG_USER} with encrypted password '${DATAWHARF_PG_PASSWORD}';"
docker exec -u postgres ${POSTGRES_CONTAINER} psql -P pager=off -U postgres -xc "grant all privileges on database ${DATAWHARF_PG_DB} to ${DATAWHARF_PG_USER};"

sed -i "s/POSTGRESHOST=.*/POSTGRESHOST=${POSTGRES_CONTAINER}/" config_demo.txt
sed -i "s/POSTGRESID=.*/POSTGRESID=${DATAWHARF_PG_USER}/" config_demo.txt
sed -i "s/POSTGRESPASSWORD=.*/POSTGRESPASSWORD=${DATAWHARF_PG_PASSWORD}/" config_demo.txt
sed -i "s/POSTGRESDATABASE=.*/POSTGRESDATABASE=${DATAWHARF_PG_DB}/" config_demo.txt

docker build . -f Dockerfile_Demo_Using_DockerHub_Ubuntu_22_04 -t datawharf_demo_using_dockerhub_baseimage:latest
docker run -d -p 8080:80 --link $POSTGRES_CONTAINER --name ${DATAWHARF_CONTAINER} datawharf_demo_using_dockerhub_baseimage:latest

echo "In 15 sec Open in browser http://localhost:8080/"
sleep 15
echo "errors ORM.SchemaAndDataErrorLog"
docker exec  -u postgres ${POSTGRES_CONTAINER} psql  -P pager=off -U postgres ${DATAWHARF_PG_DB} -xc 'select * from "ORM"."SchemaAndDataErrorLog"'

echo "I have got
Error Occurred
gcc release 07/31/2023 04:56:14 - Current Time: 08/04/2023
Error: Bound error
operation: array access
[ 1] = H {=>}
[ 2] = C
Called From: hb_orm_sqlconnect.prg, HB_ORM_SQLCONNECT:CASEFIELDNAME, line: 2795
Called From: hb_orm_sqldata.prg, HB_ORM_SQLDATA:FIELDSET, line: 187
Called From: hb_orm_sqldata.prg, HB_ORM_SQLDATA:FIELD, line: 211
Called From: src/DataWharf.prg, MYFCGI:ONFIRSTREQUEST, line: 186
Called From: /src/Harbour_FastCGI//hb_fcgi/hb_fcgi.prg, MYFCGI:WAIT, line: 282
Called From: src/DataWharf.prg, MAIN, line: 45
FastCGIRunLog.pk = 0
"
#end of script

Can you tell me what the problem is?

Thank you 
best regards
Jan

**Comment by EricLendvai** [comment]
Hello Jan, I also had problems with PostgreSQL in docker. Additionally you would need to store the database on a docker volume, to ensure you don't loose your information. 
You must ensure the account you log in to PostgreSQL with is a superuser or owns the entire database. You can always remove the orm schema, and DataWharf will rebuild it. 
I can be available next week via Skype if you want to connect. Search me as "Eric Lendvai". What country / timezone are you in?

**Reply by janbenda** [reply]
Hello Eric,
thank you for reply, after changing the user to postgres, the schema was created complete, for PG15 too. 
For persistent data i am using volume, script attached
I live in CEST.
[start_dw.sh.zip](https://github.com/EricLendvai/DataWharf/files/12342215/start_dw.sh.zip)


have a nice day
Jan
